src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
Thanks for your feedback Daniel. I'm not saying that Javaslang should be just like other libraries, it was just a suggestion based on other examples. But I'm totally fine with your ideas on this. I see your point that `pure` and `point` are just constructors. 

Here's where I'm really stuck though, and why in my previous post I suggested possibly doing away with the `Applicative` interface altogether. I wasn't just complaining about Java's type system, I know it's strengths and weaknesses. I was simply pointing out that because of its lack of higher kinded types, I'm not sure how to make the `Applicative` interface work for different kinds of applicative types. I hope I can make sense in what I'm trying to say, please let me know if I'm not.

Looking at scalaz, they have `trait Apply[F[_]]`. And then ap is defined as `def ap[A,B](fa: => F[A])(f: => F[A => B]): F[B]`. `Apply`'s type parameter describes `Validation` (or any other implementation of Apply), not the type held by `Validation`. 

We can't model this in Java. Currently my interface is defined as `public interface Applicative<T>`, and apply as `<U> Applicative<U> apply(Applicative<? extends Function<? super T,? extends U>> f)`. But this doesn't work. Looking at `Validation`, it's defined as `public interface Validation<E,T> extends Applicative<Validation<E,T>>`. That means that the overridden apply method would look like `<U> Applicative<U> apply(Applicative<? extends Function<? super Validation<E, T>, ? extends U>> f)`. But this isn't right. I need it to take a function of `Function<T,U>`, not `Function<Validation<E,T>, U>`. This is all messed up because of lack of higher kinded types. 

I may be totally missing the mark here, which is why I'm asking for your input. I'm not seeing how to make the `Applicative` interface generic enough for any kind of type that wants to implement it. Which is why I was suggesting that if we don't really think we'll add any other type to Javaslang that would implement `Applicative`, we just get rid of it complete and pull all of our applicative login into `Validation`. I really hope I'm making sense here. :) Thanks for your patience.